core/aws_ddk_core/resources/_utils.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import platform
+from typing import Optional
+
+import aws_cdk.aws_lambda as lmbda
+import boto3
+from constructs import Construct
+
+AWS_SDK_PANDAS_ARTIFACT_ACCOUNT_ID = "336392948345"
+MAX_VERSION_POLL = 50
+LAYER_PREFIXES = ["AWSDataWrangler", "AWSSDKPandas"]
+logging.basicConfig()
+logger = logging.getLogger("aws-sdk-pandas-layer-lookup")
+
+
+def _get_python_version() -> str:
+    return "Python" + "".join(platform.python_version().split(".")[0:2])
+
+
+def _latest_layer(boto3_client: boto3.client, region: str ="us-east-1") -> str:
+    layer_arn = (
+        f"arn:aws:lambda:{region}:{AWS_SDK_PANDAS_ARTIFACT_ACCOUNT_ID}:layer:AWSSDKPandas-{_get_python_version()}"
+    )
+    logger.debug(f" Layer Arn: {layer_arn}")
+    logger.debug(" Scanning versions for latest...")
+    latest = False
+    version = 1
+    while not latest:
+        try:
+            response = boto3_client.get_layer_version(
+                LayerName=layer_arn,
+                VersionNumber=version,
+            )
+            description = response["Description"]
+            version += 1
+        except Exception:
+            latest = True
+        latest_version = version - 1
```

Comment:
Unfortunately we have skipped versions in some regions... with newer `AWSSDKPandas` layers I think an example is in `us-west-2` that starts from version `2`. Not sure if this would handle this case

Reply:
Ahh yes, for the latest lookup then maybe i'll need to build in a reasonable buffer range to scan past the empty versions to check for futher versions.